code/game/machinery/recharger.dm

```diff
@@ -163,24 +163,29 @@
 			if(B.cell)
 				B.cell.charge = 0
 
+/obj/machinery/recharger/update_appearance(updates)
+	. = ..()
+	if((machine_stat & (NOPOWER|BROKEN)) || panel_open || !anchored)
+		luminosity = 0
+		return
+	luminosity = 1
+
 /obj/machinery/recharger/update_overlays()
 	. = ..()
-	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
```

Comment:
Make sure that overlays are being properly removed by update_overlay when this proc is called. When I admin proc called update_overlays on the recharger to diagnose the other issue, it duplicated the overlays. This is probably a bad thing.

![image](https://user-images.githubusercontent.com/24975989/107249685-7e4c4e00-6a2b-11eb-893b-6a56e8ff6593.png)

Reply:
@Timberpoes As far as I can tell it's always been assumed that update_overlays would not be directly called. It never removed the normal overlays either.

Reply:
See the definition for the base of update_icon on master.

Reply:
Though, in this case I think I can move the vis overlay removal into the proc base.